subsys/shell/shell.c

```diff
@@ -97,6 +97,19 @@ static void shell_cmd_buffer_clear(const struct shell *shell)
 	shell->ctx->cmd_buff_len = 0;
 }
 
+static void shell_pend_on_txdone(const struct shell *shell)
+{
+	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
+		k_poll(&shell->ctx->events[SHELL_SIGNAL_TXDONE], 1, K_FOREVER);
+		k_poll_signal_reset(&shell->ctx->signals[SHELL_SIGNAL_TXDONE]);
+	} else {
+		/* Blocking wait in case of bare metal. */
+		while (!shell->ctx->internal.flags.tx_rdy) {
```

Comment:
This will spin like crazy, should we call k_yield() inside the while loop?

Reply:
it as case when MULTITHREADING is disabled so k_yield() is not an option. Do we have some sleep function in kernel?